In [1]:

import sys,os
# sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))) # 添加上层目录到 sys.path
from data_process import X_test,X_train,y_test,y_train,Labels
from data_process import X_valid,y_valid,file_path
from data_process import data_process
import numpy as np
from sklearn import svm
from sklearn.metrics import classification_report, accuracy_score, confusion_matrix

# print(Labels,file_path)
clf=svm.SVC(kernel='linear',C=1.0,random_state=0,decision_function_shape='ovr')

clf.fit(X_train,y_train)
y_pred=clf.predict(X_test)
accuracy=accuracy_score(y_test,y_pred)
print('train:',accuracy_score(y_train,clf.predict(X_train)))
print('validation',accuracy_score(y_valid,clf.predict(X_valid)))
# print(classification_report(y_train,y_pred,target_names=Labels))
print('test:',accuracy)
print(classification_report(y_test,y_pred,target_names=Labels))

dataset/PPMI.mat
{'NC': 0, 'PD': 1}
random seed: 4
train: 1.0
validation 0.5963302752293578
test: 0.5688073394495413
              precision    recall  f1-score   support

          NC       0.33      0.38      0.36        34
          PD       0.70      0.65      0.68        75

    accuracy                           0.57       109
   macro avg       0.52      0.52      0.52       109
weighted avg       0.59      0.57      0.58       109



# ADNI
```
seed
0   0.4101  0.3667 0.3651
1   0.4382  0.3333  0.3651
2   0.4044  0.3333  0.3333
3   0.4325  0.2666  0.3492
4   0.3820  0.3500  0.3333
```

train: 0.4101123595505618
validation 0.36666666666666664
test: 0.36507936507936506

---

random seed: 1
train: 0.43820224719101125
validation 0.3333333333333333
test: 0.36507936507936506

---

train: 0.4044943820224719
validation 0.3333333333333333
test: 0.3333333333333333

------

random seed: 3
['AD', 'MCI', 'MCIn', 'MCIp', 'NC'] dataset/ADNI.mat
train: 0.43258426966292135
validation 0.26666666666666666
test: 0.3492063492063492


------

random seed: 4
['AD', 'MCI', 'MCIn', 'MCIp', 'NC'] dataset/ADNI.mat
train: 0.38202247191011235
validation 0.35
test: 0.3333333333333333

# PPMI

```
seed
0   1.0  0.6147  0.5780
1   1.0  0.5780  0.6422
2   1.0  0.5963  0.5963
3   1.0  0.4954  0.5963
4   1.0  0.5963  0.5688
```

random seed: 0
train: 1.0
validation 0.6146788990825688
test: 0.5779816513761468

------

random seed: 1
train: 1.0
validation 0.5779816513761468
test: 0.6422018348623854

------
2
train: 1.0
validation 0.5963302752293578
test: 0.5963302752293578

------

random seed: 3
train: 1.0
validation 0.4954128440366973
test: 0.5963302752293578

------

random seed: 4
train: 1.0
validation 0.5963302752293578
test: 0.5688073394495413